src/librustc_codegen_llvm/attributes.rs

```diff
@@ -84,9 +84,22 @@ pub fn set_instrument_function(cx: &CodegenCx<'ll, '_>, llfn: &'ll Value) {
     if cx.sess().instrument_mcount() {
         // Similar to `clang -pg` behavior. Handled by the
         // `post-inline-ee-instrument` LLVM pass.
+
+        // The function name varies on platforms.
+        // See test/CodeGen/mcount.c in clang.
+        let mcount_name = if cfg!(target_os = "netbsd") {
+            const_cstr!("__mcount")
+        } else if cfg!(any(
+                target_arch = "mips", target_arch = "mips64",
+                target_arch = "powerpc", target_arch = "powerpc64")) {
+            const_cstr!("_mcount")
+        } else {
+            const_cstr!("mcount")
+        };
```

Comment:
This does not seem to cover it.

Looking at clang’s `this->MCountName` instances:

* `darwin` uses `\01mcount`;
* On Aarch64 if abi is gnueabi, `\01_mcount` is used;
* On ARM-linux or ARM-unknown if abi is gnueabi `\01__gnu_mcount_nc` is used and `\01mcount` if not gnueabi;

Please make this a target property (specified in target files) and use that property here.

Reply:
(I used github search [like this](https://github.com/llvm-mirror/clang/search?q=this-%3EMCountName&unscoped_MCountName). I’m sure it missed some cases as well)